src/som/interpreter/nodes/literals/ObjectLiteralNode.java

```diff
@@ -0,0 +1,52 @@
+package som.interpreter.nodes.literals;
+
+import com.oracle.truffle.api.RootCallTarget;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.source.SourceSection;
+
+import som.compiler.MixinDefinition;
+import som.interpreter.InliningVisitor;
+import som.interpreter.nodes.ExpressionNode;
+import som.interpreter.nodes.MessageSendNode.AbstractMessageSendNode;
+import som.vmobjects.SClass;
+import som.vmobjects.SObjectWithClass;
+
+
+public class ObjectLiteralNode extends LiteralNode {
+
+  private MixinDefinition mixinDef;
+  private RootCallTarget superClassResolver;
+
+  @Child private ExpressionNode outerRead;
+  @Child private AbstractMessageSendNode newMessage;
+
+  public ObjectLiteralNode(final MixinDefinition mixinDef, final ExpressionNode outerRead,
+      final ExpressionNode newMessage, final SourceSection source) {
+    super(source);
+    this.mixinDef = mixinDef;
+    this.outerRead = outerRead;
+    this.newMessage = (AbstractMessageSendNode) newMessage;
+    this.superClassResolver = mixinDef.getSuperclassAndMixinResolutionInvokable().createCallTarget();
+  }
+
+  @Override
+  public Object executeGeneric(final VirtualFrame frame) {
+    SObjectWithClass outer = (SObjectWithClass) outerRead.executeGeneric(frame);
+    Object superclassAndMixins = superClassResolver.call(outer);
```

Comment:
This should probably be cached, similar to normal object instantiation.
Was there a specific reason why you didn't reuse the node that's doing the instantiation?